libxrdp/xrdp_caps.c

```diff
@@ -422,6 +422,20 @@ xrdp_caps_process_input(struct xrdp_rdp *self, struct stream *s,
     {
         self->client_info.use_fast_path &= ~2;
     }
+
+    // We always advertise Unicode support, so if the client supports it too,
+    // we can use it
+    if ((inputFlags & INPUT_FLAG_UNICODE) != 0)
+    {
+        self->client_info.unicode_input_support = UIS_SUPPORTED;
```

Comment:
In my client, resizing the window repeatedly triggers the function xrdp_caps_process_input. The original UIS_ACTIVE state is reset to UIS_SUPPORTED, which results in subsequent failure to deliver.

Reply:
Thanks @poetic-edge - that's a good spot.

The way we're currently doing client resizes involves using the RDP [Deactivation-Reactivation Sequence](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/dfc234ce-481a-4674-9a5d-2a7bafb14432)

Can you try this patch? I've run out of time today.

```diff
--- a/libxrdp/xrdp_caps.c
+++ b/libxrdp/xrdp_caps.c
@@ -424,16 +424,23 @@ xrdp_caps_process_input(struct xrdp_rdp *self, struct stream *s,
     }
 
     // We always advertise Unicode support, so if the client supports it too,
-    // we can use it
-    if ((inputFlags & INPUT_FLAG_UNICODE) != 0)
+    // we can use it.
+    //
+    // If Unicode support is already active, the CAPSTYPE_INPUT
+    // PDU has been received as part of a Deactivation-Reactivation sequence.
+    // In this case, ignore the flag.
+    if (self->client_info.unicode_input_support != UIS_ACTIVE)
     {
-        self->client_info.unicode_input_support = UIS_SUPPORTED;
-        LOG(LOG_LEVEL_INFO, "Client supports Unicode input");
-    }
-    else
-    {
-        self->client_info.unicode_input_support = UIS_UNSUPPORTED;
-        LOG(LOG_LEVEL_INFO, "Client does not support Unicode input");
+        if ((inputFlags & INPUT_FLAG_UNICODE) != 0)
+        {
+            self->client_info.unicode_input_support = UIS_SUPPORTED;
+            LOG(LOG_LEVEL_INFO, "Client supports Unicode input");
+        }
+        else
+        {
+            self->client_info.unicode_input_support = UIS_UNSUPPORTED;
+            LOG(LOG_LEVEL_INFO, "Client does not support Unicode input");
+        }
     }
 
     return 0;
```

Reply:
Hmm, the translation I provided earlier was not quite accurate, and it even made things worse. The initial state is not UIS_ACTIVE, and the state cannot be changed at all. I suggest moving the check for UIS_ACTIVE inside the "if ((inputFlags & INPUT_FLAG_UNICODE) != 0)" statement.

```c
if ((inputFlags & INPUT_FLAG_UNICODE) != 0)
{
    if(self->client_info.unicode_input_support != UIS_ACTIVE){
        self->client_info.unicode_input_support = UIS_SUPPORTED;
        LOG(LOG_LEVEL_INFO, "Client supports Unicode input");
    }
}
else
{
    self->client_info.unicode_input_support = UIS_UNSUPPORTED;
    LOG(LOG_LEVEL_INFO, "Client does not support Unicode input");
}
```

Reply:
Thank you. It is now functioning properly.

Reply:
Okay. I'll apply the patch.